server/db/connection.js

```diff
@@ -135,31 +138,88 @@ export class Db {
     if (isStudent == null) {
       throw new Error("User is not a student");
     }
-
     return await Course.find({ student_ids: { $in: userId } });
   }
+  
+  async getBulkCourseDetailsTeacherOnlyByIds(courseIds) {
+    const courses = courseIds.map(async (courseId) => {
+      return this.getCourseDetailsWithTeacherOnlyById(courseId)
+    })
+    return await Promise.all(courses)
+  }
 
-  async getCourseDetails(courseId) {
-    const isCourse = await Course.findOne({ _id: courseId });
-    if (isCourse == null) {
-      throw new Error("Course not found");
+  async getCourseDetailsWithTeacherOnlyById(courseId) {
+    const course = await Course.findOne({ _id: courseId });
+    if (course == null) {
+      return Promise.reject("No course found")
     }
+    const teacher = await User.findOne({
+      user_id: course.prof_id,
+      role: "teacher",
+    }, { pw: 0 });
+    return Promise.resolve(
+      {
+        course: course,
+        teacher: teacher
+      }
+    )
+  }
+
+  async getBulkCourseDetailsByIds(courseIds) {
+    const courses = courseIds.map(async (courseId) => {
+      return this.getCourseDetailsById(courseId)
+    })
+    return await Promise.all(courses)
+  }
+
+  async getCoursesStudentNotEnrolledIn(user_id) {
+    const courses = await Course.find({ student_ids: { $nin: [user_id] } })
+    const coursesWithTeachers = courses.map(async (course) => {
+      const teacher = await this.getTeacher(course.prof_id)
+      return Promise.resolve({
```

Comment:
I don't believe you need the resolve wrapper here since it's a promise by default